latest/root/usr/share/container-scripts/mongodb/db/replset/common.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+source ${CONTAINER_SCRIPTS_PATH:-}/db/common.sh
+
+# @public  Checks environmental variables for initializing a mongo replica set
+function check_repl_env_vars() {
+    # For triggering proper usage()
+    REPLICATION=1
+    [[ -v MONGODB_KEYFILE_VALUE && -v MONGODB_REPLICA_NAME ]] || usage "MONGODB_KEYFILE_VALUE and MONGODB_REPLICA_NAME have to be set"
+
+    check_db_env_vars
+}
+readonly -f check_repl_env_vars
+
+# @public Create either a replica set or add member in the background.
+#
+# @param  $1 - if this is the first member of the set, call this function to initialize. Init may differ based on what kind of replicaset
+# we are creating, for e.g (configserver rs, shard rs)
+function configure_replicaset_mode() {
+  local comm="db.isMaster().setName"
+
+  wait_for_mongo_up &>/dev/null
+
+  if [[ "$($MONGO --host $(hostfqdn) --eval "${comm}" --quiet)" =~ "${MONGODB_REPLICA_NAME:-}" ]]; then
+    log_info "Replica set '${MONGODB_REPLICA_NAME:-}' already exists, skipping initialization"
+    >/tmp/initialized
+    return 0
+  fi
+
+  if [[ "$MEMBER_ID" == "0" ]]; then
```

Comment:
If we are initializing for the first time, and this is the first member (index 0), we initialize the replica set by calling a call-back that initializes per a specific type of set - for example, a config server. 

Otherwise, we just add this as a member to an existing set.